### ISSUE TITLE
[REVIEW]: Museo ToolBox : a python library for remote sensing including a new way to handle rasters.

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1978]
**Submitting author:** @nkarasiak (<a href="http://orcid.org/0000-0002-1558-0816">Nicolas Karasiak</a>)
**Repository:** <a href="https://github.com/nkarasiak/MuseoToolBox" target ="_blank">https://github.com/nkarasiak/MuseoToolBox</a>
**Version:** v0.13.0
**Editor:** @kbarnhart
**Reviewer:** @cmillion, @mollenburger
**Archive:** <a href="https://doi.org/10.5281/zenodo.3759215" target="_blank">10.5281/zenodo.3759215</a>

## Status

[![status](https://joss.theoj.org/papers/1f4762d9910093a08034e8f4de441930/status.svg)](https://joss.theoj.org/papers/1f4762d9910093a08034e8f4de441930)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/1f4762d9910093a08034e8f4de441930"><img src="https://joss.theoj.org/papers/1f4762d9910093a08034e8f4de441930/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/1f4762d9910093a08034e8f4de441930/status.svg)](https://joss.theoj.org/papers/1f4762d9910093a08034e8f4de441930)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/nkarasiak/MuseoToolBox" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@cmillion & @mollenburger, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @kbarnhart know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/nkarasiak/MuseoToolBox">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@nkarasiak) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/nkarasiak/MuseoToolBox">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@nkarasiak) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @cmillion, @mollenburger it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.3390/rs11212512 is OK
- 10.1111/ecog.02881 is OK
- 10.1016/j.rse.2014.02.015 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01978/joss.01978/10.21105.joss.01978.pdf)

**Comment by kbarnhart** [comment]
:wave: @nkarasiak, @cmillion, @mollenburger this is the review thread for the paper. All of our communications will happen here from now on.

All reviewers have checklists at the top of this thread with the JOSS requirements. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention `openjournals/joss-reviews#1978` so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for reviews to be completed within about 2-4 weeks. Please let me know if any of you require some more time. I understand that it is late December and many people may be taking vacation and/or spending time away from work.  We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ping me (@kbarnhart) if you have any questions/concerns.

**Comment by kbarnhart** [comment]
👋 @cmillion, @mollenburger, just a quick reminder to complete your reviews. As always, if you have any questions, please feel free to ping me here.

**Comment by kbarnhart** [comment]
👋 @cmillion, @mollenburger I wanted to check in to remind you to complete your reviews. As always, if you have any questions, please feel free to ping me here.

**Comment by cmillion** [comment]
Confirming.

**Comment by kbarnhart** [comment]
@cmillion thanks for confirming.

**Comment by kbarnhart** [comment]
:wave: @cmillion, @mollenburger A friendly reminder to complete your reviews. If you have an estimated timeline, please let me know. 

@mollenburger could you please confirm receipt. 

As always, if you have any questions, please feel free to ping me here.

Thank you for being willing to review for JOSS.

**Comment by mollenburger** [comment]
confirming. I'm working on this today and hope to have it done by the end of the week.

**Comment by kbarnhart** [comment]
:wave: @cmillion, @mollenburger a friendly reminder to complete your reviews. 

If you create any issues in the submitted repository, please link to this issue by using openjournals/joss-reviews/issues/1978

When you are done, please make a note here on this issue indicating your recommendation. 

If you have any questions, please let me know.

**Comment by kbarnhart** [comment]
I wanted to check in to keep this review moving along. 

I see that there are open issues in the main repository made by @cmillion, Thank you for making issues there. 

This ping is meant as a reminder to the reviewers to make any additional issues that stand in the way of completing the review checklist, and as a reminder to @nkarasiak to address issues. 

Thanks again to all for contributing to the JOSS review process.

**Comment by kbarnhart** [comment]
:wave: another message to check in and keep this review moving forward. I see that both @cmillion and @mollenburger have made issues in the main repo (thanks 👍 ). Based on looking at them it appears that some of the issues may need to be addressed before the reviewers can complete their reviews. 

@nkarasiak can I ping you to address these issues so the reviewers can complete their reviews.

**Comment by nkarasiak** [comment]
Of course @kbarnhart, what do I need to do?
I was preparing my answers (and code/text rewriting) for the reviewers.

**Comment by kbarnhart** [comment]
thanks for the quick reply @nkarasiak.  Just addressing the issues is fine.

**Comment by kbarnhart** [comment]
👋 @nkarasiak I wanted to check in and see if you had an anticipated timeline for addressing the current set of issues.

**Comment by nkarasiak** [comment]
Hello @kbarnhart. I'm dealing with the issues, and I submitted my library to conda in order to answer to @mollenburger to ease the installation process with gdal but it takes some times...
I didn't forget to answers the issues, I just want to do it the nicest way possible :)
I hope end of next week everything will be adressed.
Kind regards,
Nicolas.

**Comment by kbarnhart** [comment]
@nkarasiak thanks for the update. It all sounds good. 

Also, sorry for the ooo-bot, I accidentally turned it on trying to demonstrate how to use it for a reviewer (and I haven't yet figured out how to turn it off). I'm not out of the office at all.

**Comment by kbarnhart** [comment]
@nkarasiak I wanted to check in on this review. Could you provide an update on your progress addressing the current set of issues raised by the reviewers?

**Comment by nkarasiak** [comment]
Hi @kbarnhart  :hand: ,
I addressed these issues several days ago : 
- https://github.com/nkarasiak/MuseoToolBox/issues/25 , I'm waiting for conda-forge to accept Museo ToolBox, but everything has been done to correct the bug @mollenburger encountered (gdal cannot be installed via pip, conda is be a great alternative to avoid this bug).
- https://github.com/nkarasiak/MuseoToolBox/issues/26 , I think everything the reviewer wanted is specified in the docstring, I specified it in the issue.

Today, I updated/corrected the paper according to suggestions from @mollenburger and @cmillion :
- https://github.com/nkarasiak/MuseoToolBox/issues/19

In any case I'm waiting for their approval (by letting them commenting or closing each issue).

**Comment by kbarnhart** [comment]
@nkarasiak thanks for the update. Perhaps you can provide another update once the conda-forge PR is merged. This will let the reviewer's know that it is sensible for them to continue with the review process.

**Comment by nkarasiak** [comment]
Here it is, my first conda-forge package :fireworks:  : https://github.com/conda-forge/museotoolbox-feedstock.
I updated the corresponding issue : https://github.com/nkarasiak/MuseoToolBox/issues/25
All the issues has been addressed, I'm now waiting for reviewers' feedback.

**Comment by arfon** [comment]
Dear authors and reviewers

We wanted to notify you that in light of the current COVID-19 pandemic, JOSS has decided to suspend submission of new manuscripts and to handle existing manuscripts (such as this one) on a "best efforts basis". We understand that you may need to attend to more pressing issues than completing a review or updating a repository in response to a review. If this is the case, a quick note indicating that you need to put a "pause" on your involvement with a review would be appreciated but is not required.

Thanks in advance for your understanding.

_Arfon Smith, Editor in Chief, on behalf of the JOSS editorial team._

**Comment by kbarnhart** [comment]
@cmillion and @mollenburger, if you are able, would you be able to complete your reviews?

**Comment by cmillion** [comment]
As soon as GDAL is added as a dependency, I'll check off "installation" and that will complete my review.

**Comment by kbarnhart** [comment]
Thanks for the update @cmillion.

@nkarasiak based on @cmillion 's comment I took a look through the repo today. I would recommend you update the readme, documentation, requirements, environment files based on some of the recent changes. For example, I couldn't see anything on the readme or documentation about installing with conda or what the difference between the three requirements files (requirements.txt, requirements-dev.txt, and environment.yml). 

@mollenburger please let me know if you have an estimated timeline for completing your review. 

Thanks to all for participating in the JOSS review process.

**Comment by nkarasiak** [comment]
Hi @kbarnhart, @cmillion  and @mollenburger ,
I updated the [develop branch](https://github.com/nkarasiak/MuseoToolBox/tree/develop) according to all your suggestions.
- The [README.md](https://github.com/nkarasiak/MuseoToolBox/blob/develop/README.md) now details the differences between `conda` and `pip`, and tell the user to install `Museo ToolBox` with `conda` to avoid the `gdal` installation on its own. As specified in the README, `gdal` can not be a `pip` dependency.
- The [CONTRIBUTING.rst](https://github.com/nkarasiak/MuseoToolBox/blob/develop/CONTRIBUTING.rst) now specifies how to install the conda environnement and how to make the local documentation and the python tests.
- The requirements/environment files are now detailed by a comment on top of each file.

Kind regards,
Nicolas.

**Comment by kbarnhart** [comment]
@nkarasiak thank you for the update and for making those changes. 

@cmillion and @mollenburger, if you are able, would you be able to complete your reviews?

**Comment by cmillion** [comment]
Done.

**Comment by kbarnhart** [comment]
@cmillion thank you for completing your review. 

@mollenburger would you be able to provide an update on when you will be able to complete you review.

**Comment by mollenburger** [comment]
I've completed the review checklist. Thanks

**Comment by kbarnhart** [comment]
@whedon generate pdf

**Comment by kbarnhart** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.3390/rs11212512 is OK
- 10.1111/ecog.02881 is OK
- 10.1016/j.rse.2014.02.015 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by kbarnhart** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01978/joss.01978/10.21105.joss.01978.pdf)

**Comment by kbarnhart** [comment]
@nkarasiak here are a few editorial comments on the paper. These comments mostly pertain to grammar, acronyms, references, and making sure the paper is suitable for a general audience. Once these are addressed, let me know and we will move forward with the manuscript process. 

- [ ] I think the Summary section would benefit from a statement describing what a raster is relative to, say, and image or an array. This statement would make the remainder of the paper more accessible for a general audience. 
- [ ] Define the acronym NDVI when first used. 
- [ ] Provide a reference for Moran's I
- [ ] Provide a reference for scikit-learn 
- [ ] There are extra spaces between text and question marks and text and colons in the main text. 
- [ ] "How can we make it as easy as possible to implement array-like functions to images?" should be "How can we make it as easy as possible to implement array-like functions on images ?" (to -> on) In addition, I note that this section will make more sense to a general audience if a statement of what an "image" is and a "raster" is can be found in the summary. 
- [ ] "The user only works with an array an confirms with a sample that the process is doing well," should be "The user only works with an array and confirms with a sample that the process is doing well," (an -> and)
- [ ] "the number of bands or the projection" should be "the number of bands, or the projection" (addition of a comma after bands)
- [ ] "The artificial intelligence (ai) module is natively built to implement scikit-learn algorithm" should be "The artificial intelligence (ai) module is natively built to implement scikit-learn algorithms" (algorithm -> algorithms)
- [ ] The three references in the section on Cross-validation should be in-text instead of parenthetical. "As stated by (Olofsson et al., 2014)" should use an in-text citation. 
- [ ] There are four undefined references in the Acknowledgments (perhaps you are calling out GitHub handles, but it is not clear)
- [ ] "For spatial cross-validation,
see (Karasiak et al., 2019) inspired from (Roberts et al., 2017)" should be "For spatial cross-validation, see Karasiak et al. (2019) inspired by Roberts et al. (2017)" (from -> by)

**Comment by nkarasiak** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01978/joss.01978/10.21105.joss.01978.pdf)

**Comment by nkarasiak** [comment]
@whedon generate pdf from branch develop

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch develop. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01978/joss.01978/10.21105.joss.01978.pdf)

**Comment by nkarasiak** [comment]
@whedon generate pdf from branch develop

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch develop. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01978/joss.01978/10.21105.joss.01978.pdf)

**Comment by nkarasiak** [comment]
Hi @kbarnhart,
Thank you for the feedback, I updated according to your suggestions. The updated paper is only on the develop branch, let me know if I have to move it to the master branch.

Kind regards,
Nicolas.

**Comment by kbarnhart** [comment]
@nkarasiak it looks to me like there are a couple of changes still remaining. 

Once you have made them, please merge these changes into the main (master) branch.

**Comment by nkarasiak** [comment]
@whedon generate pdf from branch develop

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch develop. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01978/joss.01978/10.21105.joss.01978.pdf)

**Comment by nkarasiak** [comment]
Hi @kbarnhart, sorry for the mistakes.
- I checked all the double spaces and extra spaces before question marks.
- I updated the two sentences to enhance you identified.

**Comment by kbarnhart** [comment]
@nkarasiak not a problem. There are, however, still some  extra spaces before colons in the text.

**Comment by nkarasiak** [comment]
@whedon generate pdf from branch develop

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch develop
. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01978/joss.01978/10.21105.joss.01978.pdf)

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch develop. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01978/joss.01978/10.21105.joss.01978.pdf)

**Comment by nkarasiak** [comment]
@kbarnhart, it seems like I am a big fan of spaces before colons... including in the title... But now I'm cured :)

**Comment by kbarnhart** [comment]
@nkarasiak once you have considered nkarasiak/MuseoToolBox/pull/29 and updated the repositories `master` branch could you:

- [ ] Make a tagged release of your software, and indicate the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service (e.g. figshare, an institutional repository). 
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata, this includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it); you may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here

**Comment by nkarasiak** [comment]
Hi @kbarnhart 
I updated and published the 0.13.0 version in the `master` branch :
- The git tagged version is [0.13.0](https://github.com/nkarasiak/MuseoToolBox/releases/tag/v0.13.0).
- The Zenodo archive version is also 0.13.0 with the [DOI 10.5281/zenodo.3759215](https://zenodo.org/record/3759215). Author list, title and ORCID have been added  in the archive deposit.

**Comment by kbarnhart** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.2307/2332142 is OK
- 10.3390/rs11212512 is OK
- 10.1111/ecog.02881 is OK
- 10.1016/j.rse.2014.02.015 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by kbarnhart** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01978/joss.01978/10.21105.joss.01978.pdf)

**Comment by kbarnhart** [comment]
@whedon set v0.13.0 as version

**Comment by whedon** [comment]
OK. v0.13.0 is the version.

**Comment by kbarnhart** [comment]
@whedon set 10.5281/zenodo.3759215 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3759215" target="_blank">10.5281/zenodo.3759215</a> is the archive.

**Comment by kbarnhart** [comment]
@nkarasiak I'm going to move this article to the final stage by notifying the JOSS editors in chief that it is ready to be published. They may or may note have have additional requests of you.

**Comment by kbarnhart** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.2307/2332142 is OK
- 10.3390/rs11212512 is OK
- 10.1111/ecog.02881 is OK
- 10.1016/j.rse.2014.02.015 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1426

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1426, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by arfon** [comment]
@whedon commands

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1252639377999757312) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1427
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01978](https://doi.org/10.21105/joss.01978)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by arfon** [comment]
@cmillion, @mollenburger - many thanks for your reviews here and to @kbarnhart for editing this one ✨ 

@nkarasiak - your paper is now accepted into JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01978/status.svg)](https://doi.org/10.21105/joss.01978)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01978">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01978/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01978/status.svg
   :target: https://doi.org/10.21105/joss.01978
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01978/status.svg)](https://doi.org/10.21105/joss.01978)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
